pkg/networkserver/grpc_deviceregistry.go

```diff
@@ -340,13 +345,28 @@ func (ns *NetworkServer) Set(ctx context.Context, req *ttnpb.SetEndDeviceRequest
 	if evt != nil {
 		events.Publish(evt)
 	}
-	if addDownlinkTask {
-		startAt := time.Now().UTC()
-		log.FromContext(ctx).WithField("start_at", startAt).Debug("Add downlink task")
-		if err = ns.downlinkTasks.Add(ctx, dev.EndDeviceIdentifiers, startAt, true); err != nil {
-			log.FromContext(ctx).WithError(err).Warn("Failed to add downlink task for device after set")
+
+	if !needsDownlinkCheck {
+		return dev, nil
+	}
+
+	var downAt time.Time
+	_, phy, err := getDeviceBandVersion(dev, ns.FrequencyPlans)
+	if err != nil {
+		log.FromContext(ctx).WithError(err).Warn("Failed to determine device band")
+		downAt = timeNow().UTC()
```

Comment:
Can this get us in an endless loop?

Reply:
What this does is just passes on the responsibility to figure this out to the downlink processor.
In case the `processDownlinkTask` fails to determine the band as well, a retry is performed after `downlinkRetryInterval`(currently set to 1 second).
So effectively, in case it's not possible to determine the frequency plan, the NS will retry every second until the frequency plan can be determined.

Reply:
Yeah that's what I meant with an endless loop. I think we should avoid that, and instead restart the downlink tasks when the frequency plan is set (like you do now)

Reply:
That means that if e.g. github is unreachable(or throwing unicorns) and frequency plan could not be fetched we would stop scheduling downlinks unless something triggers it.
I think that's not the right solution and can have bad consequences. Consider, for example, class C multicast device with a bunch of downlinks queued. Scheduling would only resume if a new downlink is pushed or something downlink-related is changed in the device proto via registry.
What we could do, though, would be increasing the retry interval.
Also note that it's not possible to create a device without frequency plan and we validate that it's a valid frequency plan on creation, so the value should not be the problem if errors happen here.

Reply:
OK, I see about this. I think we should have a backoff though. Frequency plans are subject to change and indeed availability may be limited when using an external source.

Reply:
I'll create an issue about it - this is out of scope of this PR

Reply:
https://github.com/TheThingsNetwork/lorawan-stack/issues/1503